gems/sorbet-runtime/test/types/configuration.rb

```diff
@@ -28,6 +28,12 @@ def self.receive(*); end
             T.let(1, String)
           end
         end
+
+        it 'T.bind raises an error' do
+          assert_raises(TypeError) do
+            T.bind(self, String)
+          end
+        end
```

Comment:
Two things:

Can you add a test that mimics more normal usage? i.e., something that actually uses instance_exec or instance_eval or something like that.

Can you also add a test using this inside a method at the top level? i.e., not inside a block:

```ruby
module Mixin1; end
module Mixin2
  def foo
    T.bind(self, T.all(Mixin1, Mixin2))
  end
end

class BadExample
  include Mixin2
end
BadExample.new.foo

class GoodExample
  include Mixin1
  include Mixin2
end
GoodExample.new.foo
```

Reply:
Added, let me know if it's the right style.